protos/sigstore_common.proto

```diff
@@ -45,22 +45,40 @@ enum HashAlgorithm {
 
 // Details of a specific public key, capturing the the key encoding method,
 // and signature algorithm.
+//
+// PublicKeyDetails captures the public key/hash algorithm combinations
+// recommended in the Sigstore ecosystem.
+//
+// This is modelled as a linear set as we want to provide a small number of
+// opinionated options instead of allowing every possible permutation.
+//
+// Any changes to this enum MUST be reflected in the algorithm registry.
+// See: docs/algorithm-registry.md
+//
 // To avoid the possibility of contradicting formats such as PKCS1 with
 // ED25519 the valid permutations are listed as a linear set instead of a
 // cartesian set (i.e one combined variable instead of two, one for encoding
 // and one for the signature algorithm).
 enum PublicKeyDetails {
         PUBLIC_KEY_DETAILS_UNSPECIFIED = 0;
         // RSA
-        PKCS1_RSA_PKCS1V5 = 1; // See RFC8017
-        PKCS1_RSA_PSS = 2; // See RFC8017
-        PKIX_RSA_PKCS1V5 = 3;
-        PKIX_RSA_PSS = 4;
+        PKCS1_RSA_PKCS1V5 = 1 [deprecated = true]; // See RFC8017
+        PKCS1_RSA_PSS = 2 [deprecated = true]; // See RFC8017
+        PKIX_RSA_PKCS1V5 = 3 [deprecated = true];
+        PKIX_RSA_PSS = 4 [deprecated = true];
+        PKIX_RSA_PKCS1_2048_SHA256 = 9;
+        PKIX_RSA_PKCS1_3072_SHA256 = 10;
+        PKIX_RSA_PKCS1_4096_SHA256 = 11;
```

Comment:
@kommendorkapten Sorry, just reviewed - Can we keep the PSS variants?

Reply:
Whoops, missed this as well. I can add those to #199, since I'm already modifying the file.

Reply:
Thanks. Do we also want to be pedantic about pkcs1 vs pkcs1v5? Also, should we support the pkcs1 encoded variants? sigstore/sigstore is able to unmarshal either pkix or pkcs1 encodings.

Reply:
(moving convo over to 199)

Reply:
Hmm, I could go either way on PKCS#1 vs. PKCS#1v1.5 -- I think it's common enough in practice to abbreviate the latter as the former. But specificity also wouldn't hurt.

In terms of encodings: I would lean towards the PKIX variants, since they're what appears in X.509. But maybe the trust root encodes them as PKCS1? I'm not sure 😅